cms/src/components/content/EditContentForm.vue

```diff
@@ -150,11 +152,47 @@ onBeforeMount(() => {
     text.value = filteredContent.text;
 });
 
+const calculateExpirationDate = (duration: string) => {
+    const publishDate = values.publishDate
+        ? DateTime.fromJSDate(new Date(values.publishDate))
+        : DateTime.now();
+    let expirationDate: DateTime = publishDate;
+
+    switch (duration) {
+        case "1 Week":
+            expirationDate = publishDate.plus({ weeks: 1 });
+            break;
+        case "2 Weeks":
+            expirationDate = publishDate.plus({ weeks: 2 });
+            break;
+        case "1 Month":
+            expirationDate = publishDate.plus({ month: 1 });
+            break;
+        case "2 Months":
+            expirationDate = publishDate.plus({ months: 2 });
+            break;
+        case "3 Months":
+            expirationDate = publishDate.plus({ months: 3 });
+            break;
+        case "6 Months":
+            expirationDate = publishDate.plus({ months: 6 });
+            break;
+        case "1 Year":
+            expirationDate = publishDate.plus({ year: 1 });
+            break;
+        default:
+            console.warn(`Unknown duration: ${duration}`);
+    }
+
+    setValues({ expiryDate: expirationDate.toISO()?.split(".")[0] as unknown as Date });
+};
+
 const save = async (validatedFormValues: typeof values, status: ContentStatus) => {
     // Make sure we don't accidentally add the 'parent' object to the content
     const contentValues = { ...validatedFormValues };
     delete contentValues["parent"];
     let publishDate;
+    let expiryDate;
```

Comment:
You're declaring `expiryDate` here, but do nothing with it, so it doesn't actually save the expiry date 😉 (try saving with an expiry date and then reload the page). You need to add the same `if` as for the publish date (the `else` is not needed I think).

```ts
if (contentValues.expiryDate) {
        publishDate = DateTime.fromJSDate(contentValues.expiryDate);
}
```
(this is needed because we handle dates generally with the Luxon library (which is where `DateTime` comes from), but veeValidate/yup deal with JavaScript dates)

Reply:
Yes, that was planned in my next commit. But thanks